Build issues in PlatformIO when using -O3

Using the latest PlatformIO (6.1.5a1) and the arduino-pico core pulled today using platform-raspberrypi in PlatformIO. 

Compiling with the following settings will throw a warning, which is reproducible using the most basic "hello world" script. It also at one point caused the USB port to become unresponsive (serial and flashing broke when not holding BOOTSEL), but I haven't been able to reproduce that. 

```
; platformio.ini
build_unflags = -Os
build_flags = -O3 -Wall
```
Build warning:
```
Compiling .pio/build/pico/FrameworkArduino/main.cpp.o
In file included from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/ArduinoAPI.h:36,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/Arduino.h:28,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.h:24,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:21:
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h: In member function 'fs::File fs::Dir::openFile(const char*)':
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h:50:7: warning: 'f.arduino::Stream::_startMillis' may be used uninitialized in this function [-Wmaybe-uninitialized]
   50 | class Stream : public Print
      |       ^~~~~~
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:241:10: note: 'f.arduino::Stream::_startMillis' was declared here
  241 |     File f(_impl->openFile(om, am), _baseFS);
      |          ^
Compiling .pio/build/pico/FrameworkArduino/posix.cpp.o
Compiling .pio/build/pico/FrameworkArduino/sdkoverride/cyw43_arch_threadsafe_background.c.o
In file included from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/ArduinoAPI.h:36,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/Arduino.h:28,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.h:24,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:21:
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h: In member function 'fs::File fs::FS::open(const arduino::String&, const char*)':
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h:50:7: warning: 'f.arduino::Stream::_startMillis' may be used uninitialized in this function [-Wmaybe-uninitialized]
   50 | class Stream : public Print
      |       ^~~~~~
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:392:10: note: 'f.arduino::Stream::_startMillis' was declared here
  392 |     File f(_impl->open(path, om, am), this);
      |          ^
Compiling .pio/build/pico/FrameworkArduino/stdlib_noniso.cpp.o
In file included from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/ArduinoAPI.h:36,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/Arduino.h:28,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.h:24,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:21:
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h: In member function 'fs::File fs::FS::open(const char*, const char*)':
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h:50:7: warning: 'f.arduino::Stream::_startMillis' may be used uninitialized in this function [-Wmaybe-uninitialized]
   50 | class Stream : public Print
      |       ^~~~~~
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:392:10: note: 'f.arduino::Stream::_startMillis' was declared here
  392 |     File f(_impl->open(path, om, am), this);
      |          ^
Compiling .pio/build/pico/FrameworkArduino/wiring_analog.cpp.o
In file included from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/ArduinoAPI.h:36,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/Arduino.h:28,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.h:24,
                 from /Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:21:
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h: In member function 'fs::File fs::File::openNextFile()':
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/api/Stream.h:50:7: warning: 'f.arduino::Stream::_startMillis' may be used uninitialized in this function [-Wmaybe-uninitialized]
   50 | class Stream : public Print
      |       ^~~~~~
/Users/reid/.platformio/packages/framework-arduinopico/cores/rp2040/FS.cpp:241:10: note: 'f.arduino::Stream::_startMillis' was declared here
  241 |     File f(_impl->openFile(om, am), _baseFS);
      |   
```

Removing the unflags will let it compile correctly, as well as changing the optimization level to `-O2` instead of `-O3`. Is this just a weird interaction between PlatformIO and the compiler, or is there some issue in the core?

--- a/cores/rp2040/FS.h
+++ b/cores/rp2040/FS.h
@@ -50,7 +50,9 @@ enum SeekMode {
 
 class File : public Stream {
 public:
-    File(FileImplPtr p = FileImplPtr(), FS *baseFS = nullptr) : _p(p), _fakeDir(nullptr), _baseFS(baseFS) { _startMillis = millis(); /* workaround -O3 spurious warning #768 */ }
+    File(FileImplPtr p = FileImplPtr(), FS *baseFS = nullptr) : _p(p), _fakeDir(nullptr), _baseFS(baseFS) {
+        _startMillis = millis(); /* workaround -O3 spurious warning #768 */
+    }
 
     // Print methods:
     size_t write(uint8_t) override;

--- a/cores/rp2040/FS.h
+++ b/cores/rp2040/FS.h
@@ -50,7 +50,7 @@ enum SeekMode {
 
 class File : public Stream {
 public:
-    File(FileImplPtr p = FileImplPtr(), FS *baseFS = nullptr) : _p(p), _fakeDir(nullptr), _baseFS(baseFS) { }
+    File(FileImplPtr p = FileImplPtr(), FS *baseFS = nullptr) : _p(p), _fakeDir(nullptr), _baseFS(baseFS) { _startMillis = millis(); /* workaround -O3 spurious warning #768 */ }
 
     // Print methods:
     size_t write(uint8_t) override;

--- a/docs/ide.rst
+++ b/docs/ide.rst
@@ -38,7 +38,7 @@ for normal operations.
 Generic RP2040 Support
 ----------------------
 If your RP2040 board isn't in the menus you can still use it with the
-IDE bu using the `Board->Generic RP2040` menu option.  You will need to
+IDE by using the `Board->Generic RP2040` menu option.  You will need to
 then set the flash size (see above) and tell the IDE how to communicate
 with the flash chip using the `Tools->Boot Stage 2` menu.
 

--- a/docs/install.rst
+++ b/docs/install.rst
@@ -5,7 +5,7 @@ The Arduino-Pico core can be installed using the Arduino IDE Boards Manager
 or using `git`.  If you want to simply write programs for your RP2040 board,
 the Boards Manager installation will suffice, but if you want to try the
 latest pre-release versions and submit improvements, you will need the `git`
-instllation.
+installation.
 
 Installing via Arduino Boards Manager
 -------------------------------------

--- a/docs/wifi.rst
+++ b/docs/wifi.rst
@@ -1,7 +1,7 @@
 WiFi (Raspberry Pi Pico W) Support
 ==================================
 
-WiFi is supported on the Raspberry Pi Pico W by selecting the "Raspbery Pi Pico W" board in the Boards Manager.  It is generally compatible with the `Arduino WiFi library <https://www.arduino.cc/en/Reference/WiFi>`__ and the `ESP8266 Arduino WiFi library <https://github.com/esp8266/Arduino>`__.
+WiFi is supported on the Raspberry Pi Pico W by selecting the "Raspberry Pi Pico W" board in the Boards Manager.  It is generally compatible with the `Arduino WiFi library <https://www.arduino.cc/en/Reference/WiFi>`__ and the `ESP8266 Arduino WiFi library <https://github.com/esp8266/Arduino>`__.
 
 Enable WiFi support by selecting the `Raspberry Pi Pico W` board in the IDE and adding ``#include <WiFi.h>`` in your sketch.
 

--- a/libraries/I2S/src/I2S.h
+++ b/libraries/I2S/src/I2S.h
@@ -74,7 +74,7 @@ public:
         return write((uint32_t)s);
     }
 
-    // Write 32 bit value to port, user responsbile for packing/alignment, etc.
+    // Write 32 bit value to port, user responsible for packing/alignment, etc.
     size_t write(int32_t val, bool sync);
 
     // Write sample to I2S port, will block until completed
@@ -83,7 +83,7 @@ public:
     size_t write24(int32_t l, int32_t r); // Note that 24b must have values left-aligned (i.e. 0xABCDEF00)
     size_t write32(int32_t l, int32_t r);
 
-    // Read 32 bit value to port, user responsbile for packing/alignment, etc.
+    // Read 32 bit value to port, user responsible for packing/alignment, etc.
     size_t read(int32_t *val, bool sync);
 
     // Read samples from I2S port, will block until data available

--- a/libraries/LEAmDNS/src/LEAmDNS.h
+++ b/libraries/LEAmDNS/src/LEAmDNS.h
@@ -1100,7 +1100,7 @@ protected:
         };
 
     public:
-        uint16_t            m_u16ID;            // Query ID (used only in lagacy queries)
+        uint16_t            m_u16ID;            // Query ID (used only in legacy queries)
         stcMDNS_RRQuestion* m_pQuestions;       // A list of queries
         uint8_t             m_u8HostReplyMask;  // Flags for reply components/answers
         bool                m_bLegacyQuery;     // Flag: Legacy query

--- a/libraries/LEAmDNS/src/LEAmDNS_Control.cpp
+++ b/libraries/LEAmDNS/src/LEAmDNS_Control.cpp
@@ -1980,7 +1980,7 @@ uint8_t MDNSResponder::_replyMaskForHost(const MDNSResponder::stcMDNS_RRHeader& 
 #ifdef MDNS_IP6_SUPPORT
             // TODO
 #endif
-        }  // Address qeuest
+        }  // Address quest
 
         stcMDNS_RRDomain hostDomain;
         if ((_buildDomainForHost(m_pcHostname, hostDomain))

--- a/libraries/WiFi/src/WiFiClientSecureBearSSL.cpp
+++ b/libraries/WiFi/src/WiFiClientSecureBearSSL.cpp
@@ -1224,7 +1224,7 @@ bool WiFiClientSecureCtx::_connectSSL(const char* hostName) {
     _x509_insecure = nullptr;
     _x509_knownkey = nullptr;
 
-    // reduce timeout after successful handshake to fail fast if server stop accepting our data for whathever reason
+    // reduce timeout after successful handshake to fail fast if server stop accepting our data for whatever reason
     if (ret) {
         _timeout = 5000;
     }

--- a/libraries/WiFi/src/wl_definitions.h
+++ b/libraries/WiFi/src/wl_definitions.h
@@ -43,7 +43,7 @@ extern "C" {
 #define WL_IPV4_LENGTH 4
 // Maximum size of a SSID list
 #define WL_NETWORKS_LIST_MAXNUM	10
-// Maxmium number of socket
+// Maximum number of socket
 #define	WIFI_MAX_SOCK_NUM	10
 // Socket not available constant
 #define SOCK_NOT_AVAIL  255

--- a/libraries/rp2040/examples/Bootsel/Bootsel.ino
+++ b/libraries/rp2040/examples/Bootsel/Bootsel.ino
@@ -1,5 +1,5 @@
 /* Simple sketch to do something on a BOOTSEL press */
-/* Releaed into the public domain */
+/* Released into the public domain */
 
 void setup() {
   Serial.begin(115200);

